internal/controller/fluxinstance_controller.go

```diff
@@ -622,3 +645,28 @@ func (r *FluxInstanceReconciler) recordMetrics(obj *fluxcdv1.FluxInstance) error
 	reporter.RecordMetrics(unstructured.Unstructured{Object: rawMap})
 	return nil
 }
+
+func (r *FluxInstanceReconciler) notify(obj *fluxcdv1.FluxInstance, reason, eventType, msg string) error {
+	notificationAddress := ""
+	if builder.ContainElementString(obj.GetComponents(), builder.MakeDefaultOptions().NotificationController) {
+		notificationAddress = fmt.Sprintf("http://%s.%s.svc.%s/",
+			builder.MakeDefaultOptions().NotificationController,
+			obj.GetNamespace(),
+			obj.GetCluster().Domain,
+		)
+	}
+
+	eventRecorder, err := events.NewRecorderForScheme(
+		r.Scheme,
+		r.EventRecorder,
+		ctrl.Log,
+		notificationAddress,
+		r.StatusManager)
+	if err != nil {
+		return err
```

Comment:
Instead of returning this error you could simply log it here since you already have `msg` to indicate what the event is for and don't need to be specific about it on every call to `notify()` like you did.